hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/freon/TestOMSnapshotDAG.java

```diff
@@ -0,0 +1,218 @@
+        snapshotInfo.getDbTxSequenceNumber());
+  }
+
+  @Test
+  void testZeroSizeKey()
+      throws IOException, InterruptedException, TimeoutException {
+
+    RandomKeyGenerator randomKeyGenerator =
+        new RandomKeyGenerator(cluster.getConf());
+    CommandLine cmd = new CommandLine(randomKeyGenerator);
+    cmd.execute("--num-of-volumes", "1",
+        "--num-of-buckets", "1",
+        "--num-of-keys", "600",
+        "--num-of-threads", "1",
+        "--key-size", "0",
+        "--factor", "THREE",
+        "--type", "RATIS",
+        "--validate-writes"
+    );
+
+    Assertions.assertEquals(600L, randomKeyGenerator.getNumberOfKeysAdded());
+    Assertions.assertEquals(600L,
+        randomKeyGenerator.getSuccessfulValidationCount());
+
+    List<OmVolumeArgs> volList = cluster.getOzoneManager()
+        .listAllVolumes("", "", 10);
+    LOG.debug("List of all volumes: {}", volList);
+    final String volumeName = volList.stream().filter(e ->
+        !e.getVolume().equals(OZONE_S3_VOLUME_NAME_DEFAULT))  // Ignore s3v vol
+        .collect(Collectors.toList()).get(0).getVolume();
+    List<OmBucketInfo> bucketList =
+        cluster.getOzoneManager().listBuckets(volumeName, "", "", 10);
+    LOG.debug("List of all buckets under the first volume: {}", bucketList);
+    final String bucketName = bucketList.get(0).getBucketName();
+
+    // Create snapshot
+    String resp = store.createSnapshot(volumeName, bucketName, "snap1");
+    LOG.debug("Snapshot created: {}", resp);
+
+    final OzoneVolume volume = store.getVolume(volumeName);
+    final OzoneBucket bucket = volume.getBucket(bucketName);
+
+    for (int i = 0; i < 6000; i++) {
+      bucket.createKey("b_" + i, 0).close();
+    }
+
+    // Create another snapshot
+    resp = store.createSnapshot(volumeName, bucketName, "snap3");
+    LOG.debug("Snapshot created: {}", resp);
+
+    // Get snapshot SST diff list
+    OzoneManager ozoneManager = cluster.getOzoneManager();
+    OMMetadataManager omMetadataManager = ozoneManager.getMetadataManager();
+    RDBStore rdbStore = (RDBStore) omMetadataManager.getStore();
+    RocksDBCheckpointDiffer differ = rdbStore.getRocksDBCheckpointDiffer();
+
+    DifferSnapshotInfo snap1 = getDifferSnapshotInfo(omMetadataManager,
+        volumeName, bucketName, "snap1");
+    DifferSnapshotInfo snap3 = getDifferSnapshotInfo(omMetadataManager,
+        volumeName, bucketName, "snap3");
+
+    // RocksDB does checkpointing in a separate thread, wait for it
+    final File checkpointSnap1 = new File(snap1.getDbPath());
+    GenericTestUtils.waitFor(checkpointSnap1::exists, 2000, 20000);
+    final File checkpointSnap3 = new File(snap3.getDbPath());
+    GenericTestUtils.waitFor(checkpointSnap3::exists, 2000, 20000);
+
+    List<String> actualDiffList = differ.getSSTDiffList(snap3, snap1);
+    LOG.debug("Got diff list: {}", actualDiffList);
+    final List<String> expectedDiffList = Arrays.asList("000059");
```

Comment:
I don't understand where "000059" comes from?  How do we know that is the filename?

Reply:
Note this integration test is **not** complete. And will be expanded in a follow-up PR.

The result is deterministic. I simply ran the test to get it. Theoretically I could do a whole RDB checkpoint open and manual SST read to find the files that holds the keyTable CF but that would be reimplementing a lot of the existing differ logic. Maybe I will do that in the follow-up jira.

Reply:
Ok, it would be good to explain in a comment so reviewers don't spend unnecessary time wondering where it came from.

Reply:
Added comment. Thanks!